modules/nf-core/nanocomp/main.nf

```diff
@@ -0,0 +1,138 @@
+process NANOCOMP {
+    label 'process_medium'
+
+    conda "bioconda:nanocomp=1.21.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/nanocomp:1.21.0--pyhdfd78af_0':
+        'quay.io/biocontainers/nanocomp:1.21.0--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(filelist)
+
+    output:
+    tuple val(meta), path("*NanoComp-report.html"), emit: report_html
+    tuple val(meta), path("*NanoComp_lengths_violin.html"), emit: lengths_violin_html
+    tuple val(meta), path("*NanoComp_log_length_violin.html"), emit: log_length_violin_html
+    tuple val(meta), path("*NanoComp_N50.html"), emit: n50_html
+    tuple val(meta), path("*NanoComp_number_of_reads.html"), emit: number_of_reads_html
+    tuple val(meta), path("*NanoComp_OverlayHistogram.html"), emit: overlay_histogram_html
+    tuple val(meta), path("*NanoComp_OverlayHistogram_Normalized.html"), emit: overlay_histogram_normalized_html
+    tuple val(meta), path("*NanoComp_OverlayLogHistogram.html"), emit: overlay_log_histogram_html
+    tuple val(meta), path("*NanoComp_OverlayLogHistogram_Normalized.html"), emit: overlay_log_histogram_normalized_html
+    tuple val(meta), path("*NanoComp_total_throughput.html"), emit: total_throughput_html
+    tuple val(meta), path("*NanoComp_quals_violin.html"), emit: quals_violin_html, optional: true
+    tuple val(meta), path("*NanoComp_OverlayHistogram_Identity.html"), emit: overlay_histogram_identity_html, optional: true
+    tuple val(meta), path("*NanoComp_OverlayHistogram_PhredScore.html"), emit: overlay_histogram_phredscore_html, optional: true
+    tuple val(meta), path("*NanoComp_percentIdentity_violin.html"), emit: percent_identity_violin_html, optional: true
+    tuple val(meta), path("*NanoComp_ActivePoresOverTime.html"), emit: active_pores_over_time_html, optional: true
+    tuple val(meta), path("*NanoComp_CumulativeYieldPlot_Gigabases.html"), emit: cumulative_yield_plot_gigabases_html, optional: true
+    tuple val(meta), path("*NanoComp_sequencing_speed_over_time.html"), emit: sequencing_speed_over_time_html, optional: true
+    tuple val(meta), path("*NanoStats.txt"), emit: stats_txt
+    path "versions.yml", emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: ""
```

Comment:
```suggestion
    def prefix = task.ext.prefix ?: "${meta.id}"
```
I'm sorry but these are the guidelines. You can still leave the code in for when someone sets the prefix to `""` so that it defaults to the non-prefix behaviour

Reply:
Would that also be an option since the tool summarises the QC of different files?

`def prefix = task.ext.prefix ?: "nanocomp"`

In the `quast` module it's also like that

Reply:
Preferably not, mainly to prevent any file name collisions later on. But you can still change the prefix yourself if you add the tool to a pipeline :)